### DISCUSSION TITLE
Private subscription channel

### DISCUSSION THREAD

**Discussion by bpobocha** [trpc/trpc#3095] [Q&A]
Hey, can anyone give me a hint on how I can make a private subscription channel on trpc and websockets? I already know the provided example, but I don't really have an idea how to use it for my purposes. 

For example, let's say I want to create a private chat between two authenticated users. What is the best way to do this? 
Thx.

**Comment by KATT** [comment]
You want a P2P thing?

**Reply by KATT** [reply]
If you're okay with it going through your servers it should already be possible

**Reply by bpobocha** [reply]
Yes, I meant sending data via websocket on my server

**Reply by draylegend** [reply]
Maybe sending an accessToken with body of a websocket message?
On a server checking the token, if a user has access to the chat
